packages/task/src/browser/task-configurations.ts

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { TaskOptions } from '../common/task-protocol';
+import { ILogger, Disposable, DisposableCollection } from '@theia/core/lib/common/';
+import URI from "@theia/core/lib/common/uri";
+import { FileSystemWatcherServer, DidFilesChangedParams, FileChange } from '@theia/filesystem/lib/common/filesystem-watcher-protocol';
+import { FileSystem } from '@theia/filesystem/lib/common';
+import * as jsoncparser from 'jsonc-parser';
+import { ParseError } from 'jsonc-parser';
+
+export interface TaskConfigurationClient {
+    /**
+     * A task has been added or removed from the known task configurations
+     * @returns an array of strings, each one being a task label
+     */
+    taskConfigurationChanged: (event: string[]) => void;
+}
+
+/**
+ * Watches potentially two tasks.json files: one in the user's home directory (in $home/.theia/ ),
+ * and  another one in the current workspace ( in $workspace/.theia/ )
+ */
+@injectable()
+export class TaskConfigurations implements Disposable {
+
+    /* Task client to send notifications to.  */
+    protected readonly toDispose = new DisposableCollection();
```

Comment:
that comment looks out-of-sync. In general it would be better to have less comments and good names. My experience is that they not so easily get out-of-sync.

Reply:
It is. Removed